keps/sig-auth/4940-psa-block-host-field-in-probes/README.md

```diff
@@ -0,0 +1,399 @@
+- [ ] (R) Graduation criteria is in place
+  - [ ] (R) [all GA Endpoints](https://github.com/kubernetes/community/pull/1806) must be hit by [Conformance Tests](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/conformance-tests.md) 
+- [ ] (R) Production readiness review completed
+- [ ] (R) Production readiness review approved
+- [ ] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] Supporting documentation—e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+<!--
+**Note:** This checklist is iterative and should be reviewed and updated every time this enhancement is being considered for a milestone.
+-->
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+We have a `Host` field that can be set from `TCPSocketAction` and
+`HTTPGetAction` fields which are part of the `ProbeHandler` and
+`LifecycleHandler` structs in Kubernetes that are used in
+`InitContainers` and `Containers` structs of `PodSpec`.
+The `Host` field is used for allowing users to specify
+another entity other than the podIP (which is the default value) to
+which Kubelet should perform probes to.
+However this opens it up for security attacks since the `Host`
+field can be set to pretty much any value in the system including
+security sensitive external hosts or localhost on the node.
+Kubelet will be probing this set `Host` value which can
+lead to blind SSRF attacks.
+
+### Goals
+
+* Add  Pod Security Admission (PSA) to enable admins to restrict
+  users from creating probes with the `Host` field set.
+* The Baseline Pod Security Standard (PSS) will be updated to enforce
+  blocking this field so that it helps with easier adoption for
+  workload operators given this is a known issue we want to prevent.
+
+### Non-Goals
+
+* Removing `.Host` field from the API and dropping support (It is
+  unsaid rule that nothing can get removed from core Kubernetes API)
+
+## Proposal
+
+There is a long term plan to deprecate the existing TCP and HTTP probe
+types in the API to replace them with ones with slightly different semantics.
+See [KEP-4559](https://github.com/kubernetes/enhancements/pull/4558) for more
+details. Given the unsolvable security problems with the Host field,
+we do not plan to offer it in the new types.
+
+Meanwhile, the older API is never going to go away. So we also want to
+add PSA to allow admins to be able to restrict users from creating
+probes with the Host field set when using the (about to be deprecated) API.
+This is implemented by [kubernetes PR 125271](https://github.com/kubernetes/kubernetes/pull/125271)
+that does exactly that.
+
```

Comment:
oops I forgot to actually write the localhost part that @BenTheElder and @danwinship pointed out here: https://github.com/kubernetes/enhancements/pull/4942#discussion_r2122082241
will do a small FUP PR to fix that up